Tests/test_content.py

```diff
@@ -534,109 +708,203 @@ def execute_testing(server, server_ip, server_version, server_numeric_version, i
+    all_tests = []
+    for test_record in tests_records:
+        test_name = test_record.get("playbookID")
+        if test_name:
+            all_tests.append(test_name)
+    return all_tests
+
+
+def manage_tests(tests_settings):
+    """
+    This function manages the execution of Demisto's tests.
+
+    Args:
+        tests_settings (TestsSettings): An object containing all the relevant data regarding how the tests should be ran
+
+    """
+    tests_settings.serverNumericVersion = get_and_print_server_numeric_version(tests_settings)
+    instances_ips = get_instances_ips_and_names(tests_settings)
+    is_nightly = tests_settings.nightly
+    server_version = tests_settings.serverVersion
+    number_of_instances = len(instances_ips)
+    prints_manager = ParallelPrintsManager(number_of_instances)
+    tests_data_keeper = TestsDataKeeper()
+
+    if tests_settings.server:
+        # If the user supplied a server - all tests will be done on that server.
+        server_ip = tests_settings.server
+        print_color("Starting tests for {}".format(server_ip), LOG_COLORS.GREEN)
+        print("Starts tests with server url - https://{}".format(server_ip))
+        all_tests = get_all_tests(tests_settings)
+        mockable_tests = []
+        print(tests_settings.specific_tests_to_run)
+        unmockable_tests = tests_settings.specific_tests_to_run if tests_settings.specific_tests_to_run else all_tests
+        execute_testing(tests_settings, server_ip, mockable_tests, unmockable_tests, tests_data_keeper, prints_manager,
+                        thread_index=0, is_ami=False)
+    elif tests_settings.isAMI:
+        """
+        Running tests in AMI configuration.
+        This is the way we run most tests, including running Circle for PRs and nightly.
+        """
+        if is_nightly:
+            """
+            If the build is a nightly build, run tests in parallel.
+            """
+            test_allocation = get_tests_allocation_for_threads(number_of_instances, tests_settings.conf_path)
+            current_thread_index = 0
+            all_unmockable_tests_list = get_unmockable_tests(tests_settings)
+            threads_array = []
+            for ami_instance_name, ami_instance_ip in instances_ips:
+                if ami_instance_name == server_version:  # Only run tests for server master
+                    current_instance = ami_instance_ip
+                    tests_allocation_for_instance = test_allocation[current_thread_index]
+
+                    unmockable_tests = [test for test in all_unmockable_tests_list if test in tests_allocation_for_instance]
+                    mockable_tests = [test for test in tests_allocation_for_instance if test not in unmockable_tests]
+                    print_color("Starting tests for {}".format(ami_instance_name), LOG_COLORS.GREEN)
+                    print("Starts tests with server url - https://{}".format(ami_instance_ip))
+
+                    if number_of_instances == 1:
```

Comment:
shouldn't this be ?
```suggestion
                    if not is_nightly or number_of_instances == 1:
```
@yaakovi take a quick look here, i want to merge this